src/evops/utils/MetricsUtils.py

```diff
@@ -0,0 +1,114 @@
+
+
+def __group_indices_by_labels(
+    labels_array: NDArray[Any, np.int32],
+) -> Dict[np.int32, NDArray[Any, np.int32]]:
+    """
+    :param labels_array: list of point cloud labels
+    :return: dictionary with labels and an array of indices belonging to this label
+    """
+    unique_labels = np.unique(labels_array)
+    dictionary = {}
+
+    for label in unique_labels:
+        label_indices = np.where(labels_array == label)[0]
+        dictionary[label] = label_indices
+
+    return dictionary
+
+
+def __are_nearly_overlapped(
+    plane_predicted: NDArray[Any, np.int32],
+    plane_gt: NDArray[Any, np.int32],
+    required_overlap: np.float64,
+) -> (bool, bool):
+    """
+    Calculate if planes are overlapped enough (required_overlap %) to be used for PP-PR metric
+    :param required_overlap: overlap threshold which will b checked to say that planes overlaps
+    :param plane_predicted: predicted segmentation
+    :param plane_gt: ground truth segmentation
+    :return: true if planes are overlapping by required_overlap % or more, false otherwise
+    """
+    intersection = np.intersect1d(plane_predicted, plane_gt)
+
+    return (
+        intersection.size / plane_predicted.size >= required_overlap
+        and intersection.size / plane_gt.size >= required_overlap,
+        intersection.size > 0,
+    )
+
+
+def __get_tp(
+    pred_labels: NDArray[Any, np.int32],
+    gt_labels: NDArray[Any, np.int32],
+    tp_condition: str,
+) -> np.int32:
+    """
+    :param pred_labels: labels of points corresponding to segmented planes
+    :param gt_labels: labels of points corresponding to ground truth planes
+    :param tp_condition: helper function to calculate statistics
+    :return: true positive received using pred_labels and gt_labels
+    """
+    true_positive = 0
+
+    unique_gt_labels = np.unique(gt_labels)
+    unique_gt_labels = np.delete(
+        unique_gt_labels,
+        np.where(unique_gt_labels == evops.metrics.constants.UNSEGMENTED_LABEL)[0],
+    )
+    assert (
+        unique_gt_labels.size != 0
+    ), "Incorrect ground truth labels unique count, most likely no labels other than UNSEGMENTED_LABEL"
+
+    unique_pred_labels = np.unique(pred_labels)
+    unique_pred_labels = np.delete(
+        unique_pred_labels,
+        np.where(unique_pred_labels == evops.metrics.constants.UNSEGMENTED_LABEL)[0],
+    )
+    assert (
+        unique_pred_labels.size != 0
+    ), "Incorrect predicted labels unique count, most likely no labels other than UNSEGMENTED_LABEL"
+
+    pred_used = set()
+    tp_condition_function = __statistics_functions[tp_condition]
+
+    for gt_label in unique_gt_labels:
+        is_already_true_positive = False
+        gt_indices = np.where(gt_labels == gt_label)[0]
+        for pred_label in unique_pred_labels:
+            pred_indices = np.where(pred_labels == pred_label)[0]
+
+            is_overlap = tp_condition_function(pred_indices, gt_indices)
+
+            if (
```

Comment:
not optimal, better to filter all cases which dont require intersection calc before it

Reply:
also, why not break after flag `is_already_true_positive` is set?